### PR TITLE
fix: provisional LEI amendment fails when country field is blank

### LINKED ISSUE #546: [Bug] Fix provisional LEI amendment failing when country field is left blank
## Summary

Amending a provisional LEI record fails with a database constraint violation whenever the **Country** field is left blank. Initial creation works fine with a blank country because the Create path already converts `""` to `NULL`. The Update path did not apply the same conversion, causing PostgreSQL to reject the write.

## Root Cause

Migration `000019_add_domain_types.up.sql` changed `lei_records.legal_address_country` to the `country_code` domain type:

```sql
CREATE DOMAIN country_code AS VARCHAR(2)
CHECK (VALUE ~ ''^[A-Z]{2}$'');
```

This domain accepts `NULL` or a valid 2-letter ISO code — empty string `""` fails the constraint.

- **Create path** (`provisionalLEIInsertPayload`): wraps the field with `nullableString()` → `""` becomes `NULL` ✅
- **Update path** (`Update()`): wrote the raw string → `""` triggers constraint violation ❌

The same inconsistency existed for `legal_jurisdiction` (no domain constraint, so no error, but semantically wrong).

## Behaviour

| Action | Country blank | Result |
|---|---|---|
| Create provisional LEI | ✓ | Succeeds — stored as `NULL` |
| Amend provisional LEI | ✓ | **Fails** — DB rejects `""` |

## Impact

No bad data was written to the database. The constraint rejected the invalid write, so the row was never modified. No data cleanup is required.

## Fix

Wrap `legal_address_country` and `legal_jurisdiction` in `nullableString()` in the Update map, matching the Create path.

**File**: `backend/internal/repository/provisional_lei_repository.go`

### PR DESCRIPTION
**Pull request by techie2000**
Fixes #546

## What changed

In `backend/internal/repository/provisional_lei_repository.go`, the `Update()` method wrote `legal_address_country` and `legal_jurisdiction` as raw strings. After migration `000019_add_domain_types.up.sql` changed `legal_address_country` to the `country_code` domain type (`CHECK (VALUE ~ '^[A-Z]{2}$')`), sending an empty string caused a PostgreSQL constraint violation — preventing any provisional LEI amendment where country was left blank.

The `Create` path already used `nullableString()` for both fields (converting `""` to `NULL`). This PR applies the same wrapper to the `Update` map.

## Changes

- `legal_address_country`: `record.LegalAddressCountry` → `nullableString(record.LegalAddressCountry)`
- `legal_jurisdiction`: `record.LegalJurisdiction` → `nullableString(record.LegalJurisdiction)` _(no DB constraint but aligns with Create path)_

## Data impact

None. The PostgreSQL constraint rejected the bad write at the database level, so no invalid values could have been stored. No migration or cleanup script is required.

## Testing

- Create a provisional LEI with no country → verify succeeds (pre-existing behaviour, unchanged)
- Amend a provisional LEI leaving country blank → verify succeeds (was failing, now fixed)
- Amend a provisional LEI with a valid 2-letter country code → verify succeeds

### CODE REVIEW

**Comment by techie2000** [comment]
## Note: Base Branch Updated

This PR's base has been updated from main to target PR #559 (eat/provisional-lei-source-notes-handling) to ensure the country/jurisdiction nullable-field fix applies on top of the LEI source+notes enhancements, maintaining focused, reviewable changes.

**Merge order:** PR #559 merges first (as base), then PR #547 on top of it.

Only the country/jurisdiction field wrapping is shown in this PR's diff.

**Comment by techie2000** [comment]
## Copilot Review: Context Update

PR 547 has been rebased to cleanly contain **only the country/jurisdiction nullable-field fix** on top of the now-merged PR 559.

**Comment status for review feedback:**

1. **Missing i18n key** (page.tsx line 1092): Addressed by PR #559 (merged ✅)
2. **Swagger docs regeneration** (service.go line 57): Addressed by PR #559 (merged ✅)  
3. **Notes field design** (service.go): Addressed by PR #559 (merged ✅)
4. **Test for UPDATE country fix** (repository.go line 148): **Still valid for PR #547** ← needs attention
5. **PR title/description mismatch**: Addressed by PR #559 (merged ✅)

**Action needed:**
- Add a regression test for the UPDATE path ensuring empty country strings convert to NULL via nullableString(), so this constraint-violation bug can't regress.

File: backend/internal/repository/provisional_lei_repository.go [hunk 1]
```diff
@@ -139,11 +140,12 @@ func (r *provisionalLEIRepository) Update(record *domain.LEIRecord) error {
...
 	result := r.db.Table("lei_raw.lei_records").Where("id = ?", record.ID).Updates(map[string]interface{}{
 		"legal_name":            record.LegalName,
-		"legal_address_country": record.LegalAddressCountry,
+		"legal_address_country": nullableString(record.LegalAddressCountry),
 		"legal_address_city":    record.LegalAddressCity,
-		"legal_jurisdiction":    record.LegalJurisdiction,
+		"legal_jurisdiction":    nullableString(record.LegalJurisdiction),
 		"entity_status":         record.EntityStatus,
 		"provisioning_source":   record.ProvisioningSource,
+		"notes":                 nullableString(record.Notes),
```
[the remark this answers is not in the document]
Comment (techie2000) [reply backend/internal/repository/provisional_lei_repository.go:153]: ✅ **Regression test added:**

Added TestProvisionalLEIUpdatePayload_ConvertsConstrainedEmptyStringsToNull in provisional_lei_repository_test.go to verify that the UPDATE path correctly converts empty country/jurisdiction strings to NULL, preventing regression of issue #546.

Also refactored Update() method to use extracted provisionalLEIUpdatePayload() helper function for consistency and improved testability.

File: frontend/app/admin/provisional-lei/page.tsx [hunk 2]
```diff
@@ -1077,6 +1080,18 @@ function ProvisionalLEIContent() {
...
+              <div>
+                <label className="block text-xs font-medium mb-1 theme-text-muted">
+                  {t('provisionalLei.form.entityStatus')}
+                </label>
+                <ThemedSelect
+                  value={createForm.entity_status}
+                  onChange={(value) => setCreateForm((f) => ({ ...f, entity_status: value }))}
+                  options={ENTITY_STATUS_OPTIONS}
+                  ariaLabel={t('provisionalLei.form.entityStatus')}
+                  className="w-full"
```
[the remark this answers is not in the document]
Comment (techie2000) [reply frontend/app/admin/provisional-lei/page.tsx:1092]: ✅ **Addressed in PR #559 (merged):**

The missing i18n key for \provisionalLei.form.entityStatus\ and all related frontend UI improvements were added in PR #559, which has since been merged to main. This PR now has access to all those translation keys.

File: backend/internal/service/provisional_lei_service.go [hunk 3]
```diff
@@ -50,6 +50,7 @@ type UpdateProvisionalLEIRequest struct {
 	LegalJurisdiction   string `json:"legal_jurisdiction"`
 	EntityStatus        string `json:"entity_status"`
 	ProvisioningSource  string `json:"provisioning_source"`
+	Notes               string `json:"notes"`
 	ParentLEI           string `json:"parent_lei"`
 	ChildLEI            string `json:"child_lei"`
 }
```
[the remark this answers is not in the document]
Comment (techie2000) [reply backend/internal/service/provisional_lei_service.go:57]: ✅ **Addressed in PR #559 (merged):**

The Swagger docs regeneration for the \
otes\ field addition to \UpdateProvisionalLEIRequest\ was handled in PR #559, which has since been merged. The docs.go file now reflects the updated schema.

File: backend/internal/service/provisional_lei_service.go [hunk 4]
```diff
@@ -183,6 +185,7 @@ func (s *provisionalLEIService) Update(lei string, req UpdateProvisionalLEIReque
 	if req.ProvisioningSource != "" {
 		record.ProvisioningSource = req.ProvisioningSource
 	}
+	record.Notes = req.Notes
```
[the remark this answers is not in the document]
Comment (techie2000) [reply backend/internal/service/provisional_lei_service.go]: ✅ **Addressed in PR #559 (merged):**

The notes field design with pointer semantics for partial update support was implemented in PR #559, which has since been merged. The field now correctly distinguishes between omitted and clear operations.

File: backend/migrations/000073_increase_provisioning_source_varchar_length.up.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+-- Error: "value too long for type character varying(50)" when inserting URLs
+-- Example: https://www.butterfieldgroup.com/sites/butterfield-corp/files/.../ (165+ chars)
+
+-- Increase provisioning_source from VARCHAR(50) to VARCHAR(1000)
+-- This accommodates:
+-- - Long document URLs (typically 100-500 chars)
+-- - References with additional metadata (up to 1000 chars)
+-- - Maintains consistency with other text fields that support URLs/references
+ALTER TABLE lei_raw.lei_records
+ALTER COLUMN provisioning_source TYPE VARCHAR(1000);
```
[the remark this answers is not in the document]
Comment (techie2000) [reply backend/migrations/000073_increase_provisioning_source_varchar_length.up.sql:12]: ✅ **Addressed in PR #559 (merged):**

The provisioning_source length increase and associated schema changes were included in PR #559, which has since been merged. This PR is now focused solely on the country/jurisdiction nullable-field fix for issue #546, which is why the title/description match the focused scope.
